noir-projects/noir-contracts/contracts/inclusion_proofs_contract/src/test.nr

```diff
@@ -0,0 +1,366 @@
+use crate::InclusionProofs;
+use dep::aztec::{
+    oracle::execution::get_contract_address,
+    prelude::AztecAddress,
+    test::helpers::{cheatcodes, test_environment::TestEnvironment},
+};
+
+global INITIAL_VALUE: Field = 69;
+
+pub unconstrained fn setup(
+    initial_value: Field,
+) -> (&mut TestEnvironment, AztecAddress, AztecAddress) {
+    // Setup env, generate keys
+    let mut env = TestEnvironment::new();
+    let owner = env.create_account(1);
+    env.impersonate(owner);
+
+    // We advance one block here, because we want to deploy our contract in block 3.
+    // This is because we will do tests later that prove the non inclusion of values and of the contract itself at block 2.
+    env.advance_block_by(1);
+
+    // Deploy contract and initialize
+    let initializer = InclusionProofs::interface().constructor(initial_value);
+    let inclusion_proofs_contract =
+        env.deploy_self("InclusionProofs").with_public_void_initializer(initializer);
+    let contract_address = inclusion_proofs_contract.to_address();
+    env.advance_block_by(1);
+    (&mut env, contract_address, owner)
+}
+
+#[test]
+unconstrained fn note_flow() {
```

Comment:
```suggestion
unconstrained fn note_validity() {
```
I think note_validity is a better name here as it communicates what is being tested.